core/src/test/scala/org/apache/spark/sql/delta/ActionSerializerSuite.scala

```diff
@@ -313,38 +336,34 @@ class ActionSerializerSuite extends QueryTest with SharedSparkSession {
   }
 
   /** Test serialization/deserialization of [[Action]] by doing an actual commit */
-  private def testActionSerDe(
-      name: String,
-      action: Action,
```

Comment:
Will it work if we just mark `action` to `action: => Action` and make all `Metadata` exposing outside `test` be `lazy val`?

Reply:
Yes. This works. let's do this instead.

e.g.
```
{
    lazy val lazyMetadata = Metadata(createdTime = Some(2222))
    testActionSerDe(
      "Metadata (with all defaults) - json serialization/deserialization",
      lazyMetadata,
      expectedJson = """{"metaData":{"id":"testId","format":{"provider":"parquet",""" +
        """"options":{}},"partitionColumns":[],"configuration":{},"createdTime":2222}}""")
  }
...
...
private def testActionSerDe(
      name: String,
      action: => Action,
...
```